statsmodels/base/transform.py

```diff
@@ -0,0 +1,169 @@
+        indicated method. The provided argument lmbda should be the lambda
+        parameter that was used to initially transform the data.
+
+        Parameters
+        ----------
+        x : array_like
+            The transformed series.
+        lmbda : float
+            The lambda parameter that was used to transform the series.
+        method : {'naive', 'normal'}
+            Indicates the method to be used in the untransformation. Defaults
+            to 'naive', which reverses the transformation. 'normal' yields an
+            optimal back-transform, assuming the transform series is normally
+            distributed.
+
+        Returns
+        -------
+        y : array_like
+            The untransformed series.
+        """
+        x = np.asarray(x)
+
+        if method == 'naive':
+            if np.isclose(lmbda, 0.):
+                y = np.exp(x)
+            else:
+                y = np.power(lmbda * x + 1, 1. / lmbda)
+        elif method == 'normal':
+            y = x  # TODO
+        else:
+            raise ValueError("Method '{0}' not understood.".format(method))
+
+        return y
+
+    def _est_lambda(self, x, bounds=(-1, 2), R=2, method='guerrero'):
+        """
+        Computes an estimate for the lambda parameter in the Box-Cox
+        transformation using method.
+
+        Parameters
+        ----------
+        x : array_like
+            The untransformed data.
+        bounds: tuple
+            Numeric 2-tuple, that indicate the solution space for the lambda
+            parameter.
+        R : int
+            The seasonality/grouping parameter.
+        method : {'guerrero', 'loglik'}
+            The method by which to estimate lambda. Defaults to 'guerrero', but
+            the profile likelihood ('loglik') is also available.
+
+        Returns
+        -------
+        lmbda : float
+            The lambda parameter.
+        """
+        if len(bounds) != 2:
+            raise ValueError("Bounds of length {0} not understood."
+                             .format(len(bounds)))
+
+        if method == 'guerrero':
+            lmbda = self._guerrero_cv(x, R, bounds)
+        elif method == 'loglik':
+            lmbda = self._loglik(x)
+        else:
+            raise ValueError("Method '{0}' not understood.".format(method))
+
+        return lmbda
+
+    def _guerrero_cv(self, x, R, bounds):
+        """
+        Computes guerrero's coefficient of variation. If no seasonality
+        is present in the data, R is set to 2 (p. 40, comment).
+
+        NOTE: Seasonality-specific auxiliaries *should* provide their own
+        seasonality parameter.
+        """
+        nobs = len(x)
+        groups = int(nobs / R)
+
+        # remove the first n < R observations from consideration.
+        grouped_data = np.reshape(x[nobs - (groups * R): nobs], (R, groups))
+
+        mean = np.mean(grouped_data, 1)
+        sd = np.std(grouped_data, 1)
```

Comment:
I could definitely write up some code to use MAD as a measure of scale, rather than the standard deviation.

(Does statsmodels currently have an IQR method?)